tools/HiveToScripts/Program.cs

```diff
@@ -0,0 +1,170 @@
+using System.Collections.Concurrent;
+using System.Text.RegularExpressions;
+
+bool doLog = true;
+
+if (args.Intersect(["--help", "-h", "help", "?"]).Any())
+{
+    Console.WriteLine(
+@"Transforms hive test execution output into sh/ps1/http scripts with big inputs dumped as separate files:
+
+    ./HiveToScripts.exe [<path-to-hive-logs> [<output-path> [<json-rpc-url>]]]
+
+Example:
+
+    ./HiveToScripts.exe T:\logs T:\output http://localhost:8551
+");
+    return;
+}
+
+string hiveLogsDirectory = args.Length > 0 ? args[0] : "logs";
+
+if (Directory.Exists(Path.Combine(hiveLogsDirectory, "logs")))
+{
+    hiveLogsDirectory = Path.Combine(hiveLogsDirectory, "logs");
+}
+
+string outputDirectory = args.Length > 1 ? args[1] : "output";
+string jsonrpcUrl = args.Length > 2 ? args[2] : "http://localhost:8551";
+
+string[] simulatorFilePaths = Directory.GetFiles(hiveLogsDirectory, "*-simulator-*.log").Select(f => Path.Combine(outputDirectory, f)).ToArray();
```

Comment:
Could it be an idea to parse the json file first for failed tests, and then use the client info id to search for the relevant requests? That way you could categorise tests under folders with a shortened test name perhaps.

Reply:
Good idea! It'll require read it two times, but also allows to ignore successful tests!